{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2979a341-b092-4e60-a8be-75c6d3f935d6",
   "metadata": {},
   "source": [
    "# Rank-based enrichment analysis of genes per topic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86e725e2-c7c3-43d8-886d-2000c9088170",
   "metadata": {},
   "source": [
    "## Load required library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "174a126a-95e5-455f-8a55-75470b28bf13",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gseapy as gp\n",
    "import matplotlib.pyplot as plt\n",
    "from pathlib import Path\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "801841cc-5dd1-461a-9036-5891b6d42ba9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.10.8'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gp.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff5a585f-7b5a-4e08-a84a-36953eb085d2",
   "metadata": {},
   "source": [
    "## Select appropriate databases from the list of databases available in Enrichr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "54d4b945-1600-4fc4-81ad-04dc0e5fa15f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(197,\n",
       " ['ARCHS4_Cell-lines',\n",
       "  'ARCHS4_IDG_Coexp',\n",
       "  'ARCHS4_Kinases_Coexp',\n",
       "  'ARCHS4_TFs_Coexp',\n",
       "  'ARCHS4_Tissues',\n",
       "  'Achilles_fitness_decrease',\n",
       "  'Achilles_fitness_increase',\n",
       "  'Aging_Perturbations_from_GEO_down',\n",
       "  'Aging_Perturbations_from_GEO_up',\n",
       "  'Allen_Brain_Atlas_10x_scRNA_2021',\n",
       "  'Allen_Brain_Atlas_down',\n",
       "  'Allen_Brain_Atlas_up',\n",
       "  'Azimuth_Cell_Types_2021',\n",
       "  'BioCarta_2013',\n",
       "  'BioCarta_2015',\n",
       "  'BioCarta_2016',\n",
       "  'BioPlanet_2019',\n",
       "  'BioPlex_2017',\n",
       "  'CCLE_Proteomics_2020',\n",
       "  'CORUM',\n",
       "  'COVID-19_Related_Gene_Sets',\n",
       "  'COVID-19_Related_Gene_Sets_2021',\n",
       "  'Cancer_Cell_Line_Encyclopedia',\n",
       "  'CellMarker_Augmented_2021',\n",
       "  'ChEA_2013',\n",
       "  'ChEA_2015',\n",
       "  'ChEA_2016',\n",
       "  'Chromosome_Location',\n",
       "  'Chromosome_Location_hg19',\n",
       "  'ClinVar_2019',\n",
       "  'DSigDB',\n",
       "  'Data_Acquisition_Method_Most_Popular_Genes',\n",
       "  'DepMap_WG_CRISPR_Screens_Broad_CellLines_2019',\n",
       "  'DepMap_WG_CRISPR_Screens_Sanger_CellLines_2019',\n",
       "  'Descartes_Cell_Types_and_Tissue_2021',\n",
       "  'DisGeNET',\n",
       "  'Disease_Perturbations_from_GEO_down',\n",
       "  'Disease_Perturbations_from_GEO_up',\n",
       "  'Disease_Signatures_from_GEO_down_2014',\n",
       "  'Disease_Signatures_from_GEO_up_2014',\n",
       "  'DrugMatrix',\n",
       "  'Drug_Perturbations_from_GEO_2014',\n",
       "  'Drug_Perturbations_from_GEO_down',\n",
       "  'Drug_Perturbations_from_GEO_up',\n",
       "  'ENCODE_Histone_Modifications_2013',\n",
       "  'ENCODE_Histone_Modifications_2015',\n",
       "  'ENCODE_TF_ChIP-seq_2014',\n",
       "  'ENCODE_TF_ChIP-seq_2015',\n",
       "  'ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X',\n",
       "  'ESCAPE',\n",
       "  'Elsevier_Pathway_Collection',\n",
       "  'Enrichr_Libraries_Most_Popular_Genes',\n",
       "  'Enrichr_Submissions_TF-Gene_Coocurrence',\n",
       "  'Enrichr_Users_Contributed_Lists_2020',\n",
       "  'Epigenomics_Roadmap_HM_ChIP-seq',\n",
       "  'FANTOM6_lncRNA_KD_DEGs',\n",
       "  'GO_Biological_Process_2013',\n",
       "  'GO_Biological_Process_2015',\n",
       "  'GO_Biological_Process_2017',\n",
       "  'GO_Biological_Process_2017b',\n",
       "  'GO_Biological_Process_2018',\n",
       "  'GO_Biological_Process_2021',\n",
       "  'GO_Cellular_Component_2013',\n",
       "  'GO_Cellular_Component_2015',\n",
       "  'GO_Cellular_Component_2017',\n",
       "  'GO_Cellular_Component_2017b',\n",
       "  'GO_Cellular_Component_2018',\n",
       "  'GO_Cellular_Component_2021',\n",
       "  'GO_Molecular_Function_2013',\n",
       "  'GO_Molecular_Function_2015',\n",
       "  'GO_Molecular_Function_2017',\n",
       "  'GO_Molecular_Function_2017b',\n",
       "  'GO_Molecular_Function_2018',\n",
       "  'GO_Molecular_Function_2021',\n",
       "  'GTEx_Aging_Signatures_2021',\n",
       "  'GTEx_Tissue_Expression_Down',\n",
       "  'GTEx_Tissue_Expression_Up',\n",
       "  'GWAS_Catalog_2019',\n",
       "  'GeneSigDB',\n",
       "  'Gene_Perturbations_from_GEO_down',\n",
       "  'Gene_Perturbations_from_GEO_up',\n",
       "  'Genes_Associated_with_NIH_Grants',\n",
       "  'Genome_Browser_PWMs',\n",
       "  'HDSigDB_Human_2021',\n",
       "  'HDSigDB_Mouse_2021',\n",
       "  'HMDB_Metabolites',\n",
       "  'HMS_LINCS_KinomeScan',\n",
       "  'HomoloGene',\n",
       "  'HuBMAP_ASCT_plus_B_augmented_w_RNAseq_Coexpression',\n",
       "  'HuBMAP_ASCTplusB_augmented_2022',\n",
       "  'HumanCyc_2015',\n",
       "  'HumanCyc_2016',\n",
       "  'Human_Gene_Atlas',\n",
       "  'Human_Phenotype_Ontology',\n",
       "  'InterPro_Domains_2019',\n",
       "  'Jensen_COMPARTMENTS',\n",
       "  'Jensen_DISEASES',\n",
       "  'Jensen_TISSUES',\n",
       "  'KEA_2013',\n",
       "  'KEA_2015',\n",
       "  'KEGG_2013',\n",
       "  'KEGG_2015',\n",
       "  'KEGG_2016',\n",
       "  'KEGG_2019_Human',\n",
       "  'KEGG_2019_Mouse',\n",
       "  'KEGG_2021_Human',\n",
       "  'Kinase_Perturbations_from_GEO_down',\n",
       "  'Kinase_Perturbations_from_GEO_up',\n",
       "  'L1000_Kinase_and_GPCR_Perturbations_down',\n",
       "  'L1000_Kinase_and_GPCR_Perturbations_up',\n",
       "  'LINCS_L1000_Chem_Pert_down',\n",
       "  'LINCS_L1000_Chem_Pert_up',\n",
       "  'LINCS_L1000_Ligand_Perturbations_down',\n",
       "  'LINCS_L1000_Ligand_Perturbations_up',\n",
       "  'Ligand_Perturbations_from_GEO_down',\n",
       "  'Ligand_Perturbations_from_GEO_up',\n",
       "  'MAGMA_Drugs_and_Diseases',\n",
       "  'MCF7_Perturbations_from_GEO_down',\n",
       "  'MCF7_Perturbations_from_GEO_up',\n",
       "  'MGI_Mammalian_Phenotype_2013',\n",
       "  'MGI_Mammalian_Phenotype_2017',\n",
       "  'MGI_Mammalian_Phenotype_Level_3',\n",
       "  'MGI_Mammalian_Phenotype_Level_4',\n",
       "  'MGI_Mammalian_Phenotype_Level_4_2019',\n",
       "  'MGI_Mammalian_Phenotype_Level_4_2021',\n",
       "  'MSigDB_Computational',\n",
       "  'MSigDB_Hallmark_2020',\n",
       "  'MSigDB_Oncogenic_Signatures',\n",
       "  'Microbe_Perturbations_from_GEO_down',\n",
       "  'Microbe_Perturbations_from_GEO_up',\n",
       "  'Mouse_Gene_Atlas',\n",
       "  'NCI-60_Cancer_Cell_Lines',\n",
       "  'NCI-Nature_2015',\n",
       "  'NCI-Nature_2016',\n",
       "  'NIH_Funded_PIs_2017_AutoRIF_ARCHS4_Predictions',\n",
       "  'NIH_Funded_PIs_2017_GeneRIF_ARCHS4_Predictions',\n",
       "  'NIH_Funded_PIs_2017_Human_AutoRIF',\n",
       "  'NIH_Funded_PIs_2017_Human_GeneRIF',\n",
       "  'NURSA_Human_Endogenous_Complexome',\n",
       "  'OMIM_Disease',\n",
       "  'OMIM_Expanded',\n",
       "  'Old_CMAP_down',\n",
       "  'Old_CMAP_up',\n",
       "  'Orphanet_Augmented_2021',\n",
       "  'PFOCR_Pathways',\n",
       "  'PPI_Hub_Proteins',\n",
       "  'PanglaoDB_Augmented_2021',\n",
       "  'Panther_2015',\n",
       "  'Panther_2016',\n",
       "  'Pfam_Domains_2019',\n",
       "  'Pfam_InterPro_Domains',\n",
       "  'PheWeb_2019',\n",
       "  'PhenGenI_Association_2021',\n",
       "  'Phosphatase_Substrates_from_DEPOD',\n",
       "  'ProteomicsDB_2020',\n",
       "  'RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO',\n",
       "  'RNAseq_Automatic_GEO_Signatures_Human_Down',\n",
       "  'RNAseq_Automatic_GEO_Signatures_Human_Up',\n",
       "  'RNAseq_Automatic_GEO_Signatures_Mouse_Down',\n",
       "  'RNAseq_Automatic_GEO_Signatures_Mouse_Up',\n",
       "  'Rare_Diseases_AutoRIF_ARCHS4_Predictions',\n",
       "  'Rare_Diseases_AutoRIF_Gene_Lists',\n",
       "  'Rare_Diseases_GeneRIF_ARCHS4_Predictions',\n",
       "  'Rare_Diseases_GeneRIF_Gene_Lists',\n",
       "  'Reactome_2013',\n",
       "  'Reactome_2015',\n",
       "  'Reactome_2016',\n",
       "  'SILAC_Phosphoproteomics',\n",
       "  'SubCell_BarCode',\n",
       "  'SysMyo_Muscle_Gene_Sets',\n",
       "  'TF-LOF_Expression_from_GEO',\n",
       "  'TF_Perturbations_Followed_by_Expression',\n",
       "  'TG_GATES_2020',\n",
       "  'TRANSFAC_and_JASPAR_PWMs',\n",
       "  'TRRUST_Transcription_Factors_2019',\n",
       "  'Table_Mining_of_CRISPR_Studies',\n",
       "  'Tabula_Sapiens',\n",
       "  'TargetScan_microRNA',\n",
       "  'TargetScan_microRNA_2017',\n",
       "  'Tissue_Protein_Expression_from_Human_Proteome_Map',\n",
       "  'Tissue_Protein_Expression_from_ProteomicsDB',\n",
       "  'Transcription_Factor_PPIs',\n",
       "  'UK_Biobank_GWAS_v1',\n",
       "  'Virus-Host_PPI_P-HIPSTer_2020',\n",
       "  'VirusMINT',\n",
       "  'Virus_Perturbations_from_GEO_down',\n",
       "  'Virus_Perturbations_from_GEO_up',\n",
       "  'WikiPathway_2021_Human',\n",
       "  'WikiPathways_2013',\n",
       "  'WikiPathways_2015',\n",
       "  'WikiPathways_2016',\n",
       "  'WikiPathways_2019_Human',\n",
       "  'WikiPathways_2019_Mouse',\n",
       "  'dbGaP',\n",
       "  'huMAP',\n",
       "  'lncHUB_lncRNA_Co-Expression',\n",
       "  'miRTarBase_2017'])"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human = gp.get_library_name(organism='Human')\n",
    "len(human), human#[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "573eac8d-7832-4181-94b9-cccb8703a0d7",
   "metadata": {},
   "source": [
    "## Load gene and ranks data topic wise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9a1e5971-b7c4-44b5-97de-8ae0262cc416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>topic_0</th>\n",
       "      <th>topic_0_prop</th>\n",
       "      <th>topic_1</th>\n",
       "      <th>topic_1_prop</th>\n",
       "      <th>topic_2</th>\n",
       "      <th>topic_2_prop</th>\n",
       "      <th>topic_3</th>\n",
       "      <th>topic_3_prop</th>\n",
       "      <th>topic_4</th>\n",
       "      <th>topic_4_prop</th>\n",
       "      <th>topic_5</th>\n",
       "      <th>topic_5_prop</th>\n",
       "      <th>topic_6</th>\n",
       "      <th>topic_6_prop</th>\n",
       "      <th>topic_7</th>\n",
       "      <th>topic_7_prop</th>\n",
       "      <th>topic_8</th>\n",
       "      <th>topic_8_prop</th>\n",
       "      <th>topic_9</th>\n",
       "      <th>topic_9_prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FTL</td>\n",
       "      <td>0.045098</td>\n",
       "      <td>CD74</td>\n",
       "      <td>0.043416</td>\n",
       "      <td>IGKC</td>\n",
       "      <td>0.357901</td>\n",
       "      <td>KRT14</td>\n",
       "      <td>0.068934</td>\n",
       "      <td>MALAT1</td>\n",
       "      <td>0.038370</td>\n",
       "      <td>S100A8</td>\n",
       "      <td>0.061923</td>\n",
       "      <td>MALAT1</td>\n",
       "      <td>0.401556</td>\n",
       "      <td>B2M</td>\n",
       "      <td>0.020044</td>\n",
       "      <td>RPL41</td>\n",
       "      <td>0.020805</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>0.045418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FTH1</td>\n",
       "      <td>0.043526</td>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>0.033774</td>\n",
       "      <td>IGLC2</td>\n",
       "      <td>0.110024</td>\n",
       "      <td>KRT5</td>\n",
       "      <td>0.014258</td>\n",
       "      <td>B2M</td>\n",
       "      <td>0.030612</td>\n",
       "      <td>S100A9</td>\n",
       "      <td>0.037184</td>\n",
       "      <td>NEAT1</td>\n",
       "      <td>0.014969</td>\n",
       "      <td>MALAT1</td>\n",
       "      <td>0.019620</td>\n",
       "      <td>RPLP1</td>\n",
       "      <td>0.018459</td>\n",
       "      <td>KRT1</td>\n",
       "      <td>0.039266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B2M</td>\n",
       "      <td>0.020449</td>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>0.033032</td>\n",
       "      <td>IGLC3</td>\n",
       "      <td>0.069180</td>\n",
       "      <td>B2M</td>\n",
       "      <td>0.013784</td>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>0.025853</td>\n",
       "      <td>KRT16</td>\n",
       "      <td>0.024092</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>0.008900</td>\n",
       "      <td>FTH1</td>\n",
       "      <td>0.012619</td>\n",
       "      <td>RPS18</td>\n",
       "      <td>0.018310</td>\n",
       "      <td>DMKN</td>\n",
       "      <td>0.017053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>0.017277</td>\n",
       "      <td>B2M</td>\n",
       "      <td>0.022879</td>\n",
       "      <td>IGHG3</td>\n",
       "      <td>0.067601</td>\n",
       "      <td>MALAT1</td>\n",
       "      <td>0.010009</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>0.013113</td>\n",
       "      <td>KRT17</td>\n",
       "      <td>0.022191</td>\n",
       "      <td>RPS27</td>\n",
       "      <td>0.005407</td>\n",
       "      <td>LGALS1</td>\n",
       "      <td>0.006454</td>\n",
       "      <td>RPL10</td>\n",
       "      <td>0.018259</td>\n",
       "      <td>LY6D</td>\n",
       "      <td>0.011399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MALAT1</td>\n",
       "      <td>0.015034</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>0.021814</td>\n",
       "      <td>IGHG1</td>\n",
       "      <td>0.047243</td>\n",
       "      <td>RPLP1</td>\n",
       "      <td>0.009915</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.007554</td>\n",
       "      <td>KRT14</td>\n",
       "      <td>0.020755</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>0.005208</td>\n",
       "      <td>S100A6</td>\n",
       "      <td>0.006210</td>\n",
       "      <td>RPL34</td>\n",
       "      <td>0.018172</td>\n",
       "      <td>NEAT1</td>\n",
       "      <td>0.011235</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  topic_0  topic_0_prop  topic_1  topic_1_prop topic_2  topic_2_prop topic_3  \\\n",
       "0     FTL      0.045098     CD74      0.043416    IGKC      0.357901   KRT14   \n",
       "1    FTH1      0.043526   TMSB4X      0.033774   IGLC2      0.110024    KRT5   \n",
       "2     B2M      0.020449  HLA-DRA      0.033032   IGLC3      0.069180     B2M   \n",
       "3  TMSB4X      0.017277      B2M      0.022879   IGHG3      0.067601  MALAT1   \n",
       "4  MALAT1      0.015034     ACTB      0.021814   IGHG1      0.047243   RPLP1   \n",
       "\n",
       "   topic_3_prop topic_4  topic_4_prop topic_5  topic_5_prop topic_6  \\\n",
       "0      0.068934  MALAT1      0.038370  S100A8      0.061923  MALAT1   \n",
       "1      0.014258     B2M      0.030612  S100A9      0.037184   NEAT1   \n",
       "2      0.013784  TMSB4X      0.025853   KRT16      0.024092  MT-CO1   \n",
       "3      0.010009    ACTB      0.013113   KRT17      0.022191   RPS27   \n",
       "4      0.009915   HLA-B      0.007554   KRT14      0.020755  MT-CO3   \n",
       "\n",
       "   topic_6_prop topic_7  topic_7_prop topic_8  topic_8_prop topic_9  \\\n",
       "0      0.401556     B2M      0.020044   RPL41      0.020805   KRT10   \n",
       "1      0.014969  MALAT1      0.019620   RPLP1      0.018459    KRT1   \n",
       "2      0.008900    FTH1      0.012619   RPS18      0.018310    DMKN   \n",
       "3      0.005407  LGALS1      0.006454   RPL10      0.018259    LY6D   \n",
       "4      0.005208  S100A6      0.006210   RPL34      0.018172   NEAT1   \n",
       "\n",
       "   topic_9_prop  \n",
       "0      0.045418  \n",
       "1      0.039266  \n",
       "2      0.017053  \n",
       "3      0.011399  \n",
       "4      0.011235  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rank_by_topic = pd.read_csv(\"Results/Rank_by_topic.csv\")\n",
    "\n",
    "rank_by_topic.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "75e45c8e-ce76-48c1-8fa0-d9557838a280",
   "metadata": {},
   "outputs": [],
   "source": [
    "rkn_DF_list = []\n",
    "for i in list(rank_by_topic):\n",
    "    if \"prop\" in i:\n",
    "        Topic, score = i.replace(\"_prop\", \"\"), i\n",
    "        df = rank_by_topic[[Topic, score]]\n",
    "        rkn_DF_list.append(df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20e161eb-1dbd-4eaf-8944-b28fb51e19a8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "a33f3421-a860-4acd-abf4-c3f19ddfc478",
   "metadata": {},
   "source": [
    "## Enrichment function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0473f4ff-aca5-4a93-b3b8-08e82f72edd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 4 µs, sys: 1 µs, total: 5 µs\n",
      "Wall time: 11.4 µs\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "Outdir = \"Results/Enrichment/\"\n",
    "\n",
    "def Enrich_topic(rkn_topic_DF_list, Gene_set, Print=False):\n",
    "    for topic_rkn_list in rkn_topic_DF_list:\n",
    "        # print(topic_rkn_list)\n",
    "        Folder = Outdir + Gene_set\n",
    "        File = Folder +\"/\"+ list(topic_rkn_list)[0] + \".tsv\"\n",
    "        Path(Folder).mkdir(parents=True, exist_ok=True)\n",
    "        # print(File, Folder)\n",
    "        try:\n",
    "            pre_res = gp.prerank(rnk=topic_rkn_list, gene_sets=Gene_set,\n",
    "                         processes=4,\n",
    "                         permutation_num=10, # reduce number to speed up testing\n",
    "                         outdir='temp/temp', format='png', seed=6)\n",
    "        \n",
    "            terms = pre_res.res2d.index\n",
    "\n",
    "            Top_10 = pre_res.res2d.head()\n",
    "            # print(type(Top_10))\n",
    "            Top_10.to_csv(File, sep=\"\\t\")\n",
    "        except:\n",
    "            print(f\"Error:{Gene_set}\")\n",
    "            return\n",
    "        \n",
    "        # break\n",
    "    print(Gene_set)\n",
    "\n",
    "    \n",
    "# Enrich_topic(rkn_DF_list, 'Azimuth_Cell_Types_2021')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7826f060-94e2-4b25-8ecc-530b98052dfb",
   "metadata": {},
   "source": [
    "## Enrichment analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0d846b9e-9c09-41a3-be18-61dd15f33cf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "Enrich_topic(rkn_DF_list, 'Azimuth_Cell_Types_2021')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "034e340e-bcb4-4301-a4df-0090e2bcf0b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "Enrich_topic(rkn_DF_list, 'Genes_Associated_with_NIH_Grants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a438e30c-95b1-4f6e-8879-fd4ca17dfb70",
   "metadata": {},
   "outputs": [],
   "source": [
    "Enrich_topic(rkn_DF_list, 'CellMarker_Augmented_2021')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cc3a313-1cd1-4b2d-a5aa-fad502415042",
   "metadata": {},
   "source": [
    "# END"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Hackin_Omics",
   "language": "python",
   "name": "hackin_omics"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
